src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java

```diff
@@ -207,7 +208,12 @@ private void sendPullRequestEvent(PullRequestEvent event, EventType eventType, b
             {
                 try
                 {
-                    pullRequestService.canMerge(pullRequest.getToRef().getRepository().getId(), pullRequest.getId());
+                    ScmPullRequestCommandFactory pullRequestCommandFactory = scmService.getPullRequestCommandFactory(pullRequest);
+                    // Require 4.10 to avoid deprecation warning.
+                    if (Version.check(applicationPropertiesService.getBuildVersion(), ">= 4.10.0"))
```

Comment:
To avoid the new dependency (which you'll have to bundle into the plugin, since we don't provide it), I'd use the [`Version`](https://developer.atlassian.com/static/javadoc/bitbucket-server/4.10.0/api/reference/com/atlassian/bitbucket/util/Version.html) class that is part of our API.

```java
boolean useCanMerge = new Version(applicationPropertiesService.getBuildVersion())).compareTo(new Version(4, 10)) < 0;
```

I'd probably do that check once, in the constructor, and assign the result to a `final` field on the class. The Bitbucket Server version can't change at runtime, so whatever version you start with is the version you'll end with. That way this code can just be:

```java
Command<?> command;
if (useCanMerge) {
    command = pullRequestCommandFactory.canMerge();
} else {
    command = pullRequestCommandFactory.tryMerge(pullRequest);
}
command.call();
```

Note that these methods return a `Command`, which you must `call()` in order to _run_ it. Otherwise it does nothing. (The `Command` allows setting timeouts, as well as requesting asynchronous execution. For this specific `Command`, though, we don't need to make any changes. You can just `call()` it.)